packages/grid_client/src/high_level/twinDeploymentHandler.ts

```diff
@@ -265,6 +265,36 @@ class TwinDeploymentHandler {
     return deployments;
   }
 
+  async checkFarmIps(twinDeployments: TwinDeployment[]) {
+    const farmIPs: Map<number, number> = new Map();
+
+    for (const twinDeployment of twinDeployments) {
+      if (twinDeployment.operation === Operations.deploy) {
```

Comment:
add one more check on the public ips if it's 0 to skip getting nodes and farms (fewer requests, better performance)